ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -70,125 +56,26 @@ export class ImportedDocuments extends React.Component {
         },
       },
       filteredRecords: null,
+      lastIndex: 0,
     };
-    this.mode = '';
-    this.status = '';
-    this.lastIndex = 0;
   }
 
-  componentDidMount() {
-    const { taskId } = this.props;
-    this.intervalId = setInterval(
-      () => this.checkForData(taskId),
-      invenioConfig.IMPORTER.fetchTaskStatusIntervalSecs
-    );
-    this.checkForData(taskId);
-  }
-
-  componentWillUnmount = () => {
-    this.intervalId && clearInterval(this.intervalId);
-  };
-
-  checkForData = async () => {
-    const { importCompleted, data } = this.state;
-    const { taskId } = this.props;
-    if (!importCompleted) {
-      const nextEntry = _get(data, 'loaded_entries', 0);
-      const response = await importerApi.check(taskId, nextEntry);
-      if (response.data.status !== 'RUNNING') {
-        this.setState({
-          importCompleted: true,
-          isLoading: false,
-          data: response.data,
-        });
-      } else {
-        this.setState({
-          isLoading: true,
-          data: response.data,
-        });
-      }
-      this.filteredRecords();
-      this.calculateMode();
-      this.calculateStatus();
-      this.calculateStatistics(response.data);
-    } else {
-      this.intervalId && clearInterval(this.intervalId);
-    }
-  };
-
-  calculateMode = () => {
-    const { data } = this.state;
-    switch (data.mode) {
-      case 'IMPORT':
-        this.mode = (
-          <Label color="blue" basic>
-            Import
-          </Label>
-        );
-        break;
-      case 'DELETE':
-        this.mode = (
-          <Label color="red" basic>
-            Delete
-          </Label>
-        );
-        break;
-      case 'PREVIEW_IMPORT':
-        this.mode = (
-          <Label color="teal" basic>
-            Preview (import)
-          </Label>
-        );
-        break;
-      case 'PREVIEW_DELETE':
-        this.mode = (
-          <Label color="teal" basic>
-            Preview (delete)
-          </Label>
-        );
-        break;
-      case 'ERROR':
-        this.mode = (
-          <Label color="red" basic>
-            Error
-          </Label>
-        );
-        break;
-      default:
-        throw Error('There was a problem processing the mode.');
-    }
-  };
+  componentDidUpdate(prevProps) {
+    const { data } = this.props;
 
-  statusLabel = () => {
-    const { data, isLoading } = this.state;
-    if (!data) return <Label color="grey">Fetching status</Label>;
-    if (isLoading)
-      return (
-        <Label color="blue">
-          <Icon loading name="circle notch" /> Importing literature
-        </Label>
-      );
-    switch (data.status) {
-      case 'SUCCEEDED':
-        return <Label color="green">Import successfull</Label>;
-      case 'CANCELLED':
-        return <Label color="yellow">Import cancelled</Label>;
-      default:
-        return <Label color="red">Import failed</Label>;
+    if (data && data !== prevProps.data) {
+      this.filterRecords();
```

Comment:
Yes, thanks @Pineirin, I would avoid as much as possible to use `componentDidUpdate` and then compare props. 
It can lead to nasty bugs.
Why is this needed? Do you need it at first run?

Reply:
i see 🤔  i did not know that. Nor have i seen it doing weird things. I need it to run certain methods after the data prop has changed